collections/_article/dev-snapshot-godot-4-2-beta-1.md

```diff
@@ -0,0 +1,376 @@
+
+For Forward+ and Mobile rendering methods this release introduces 2D HDR rendering, unlocking 3D effects, such as glow, for 2D games ([GH-80215](https://github.com/godotengine/godot/pull/80215)). HDR rendering can be used to substantially improve the quality of 2D rendering at the cost of performance. The Compatibility renderer also gets a new and anticipated feature — 3D shadows ([GH-77496](https://github.com/godotengine/godot/pull/77496)). Both of these changes were crafted by [Clay John](https://github.com/clayjohn).
+
+Last, but not least, an optional ANGLE-backed OpenGL renderer was added for macOS and Windows ([GH-72831](https://github.com/godotengine/godot/pull/72831)). ANGLE is a compatibility layer for OpenGL on top of Metal and Direct3D 11, which allows us to work around the deprecated and unmaintained OpenGL drivers on macOS, and similarly outdated OpenGL drivers on Windows for some older integrated chipsets. This should increase the portability of Godot games on lower end devices.
+
+#### Particles
+
+Our brilliant VFX champion [Ilaria Cislaghi](https://github.com/QbieShay) grew tired of a messy implementation of GPU particles and organized a rework aimed at improving internal structures for better maintenance ([GH-79527](https://github.com/godotengine/godot/pull/79527)). Along the way, she also tweaked all sorts of bits and bobs to give VFX artists more control over particle motion.
+
+It's now possible to directly animate velocity over life, inherit projectile velocity, and change the emission amount of particles. While this is a significant rework, it was done in a way that avoids compatibility breakage, so you should be able to continue using all your current particle effects with no regressions.
+
+On the more technical side of things, both GPU and CPU particles now have a dedicated signal for when they are finished with emission and simulation ([GH-76853](https://github.com/godotengine/godot/pull/76853), [GH-76859](https://github.com/godotengine/godot/pull/76859) by [HolonProduction](https://github.com/HolonProduction)). This can be extremely useful for animations that require a precise order of operations.
+
+The CPU particle system exists as a fallback for less capable and less powerful devices, but with modern hardware the need for it is significantly reduced. Specifically for the time when you decide to move to GPU particles, which offer way more functionality, especially with the aforementioned rework, there is now a built-in tool to convert CPU particles to their GPU equivalent ([GH-80779](https://github.com/godotengine/godot/pull/80779) by [Yuri Rubinsky](https://github.com/Chaosus)).
```

Comment:
```suggestion
Importantly, please note that with these changes the GPUParticles and CPUParticles will start to get out of sync. That is to say that we are no longer holding back features from GPUParticles that would be difficult to implement in CPUParticles. Now that all supported backends can support GPUParticles, we view CPUParticles as a low-cost, simplified alternative to GPUParticles, instead of a direct 1-to-1 equivalent implemented on the CPU for older devices. We encourage users to move towards using GPUParticles. To aid in that transition there is now a built-in tool to convert CPU particles to their GPU equivalent ([GH-80779](https://github.com/godotengine/godot/pull/80779) by [Yuri Rubinsky](https://github.com/Chaosus)).
```